lib/rules/require-prop-comment.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @author CZB
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+const utils = require('../utils')
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'require props to have a comment',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/require-prop-comment.html'
+    },
+    fixable: null,
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          type: { enum: ['JSDoc', 'line', 'block', 'any'] }
+        },
+        additionalProperties: false
+      }
+    ],
+    messages: {
+      requireAnyComment: 'The "{{name}}" property should have a comment.',
+      requireLineComment: 'The "{{name}}" property should have a line comment.',
+      requireBlockComment:
+        'The "{{name}}" property should have a block comment.',
+      requireJSDocComment:
+        'The "{{name}}" property should have a JSDoc comment.'
+    }
+  },
+  /** @param {RuleContext} context */
+  create(context) {
+    /** @type {{type: "JSDoc" | "line" | "block" | "any"}|undefined} */
+    const schema = context.options[0]
+    const type = schema ? schema.type : 'JSDoc'
```

Comment:
Just in case the option was set to just an object,
I think it's better to do something like this:

```suggestion
    const type = (schema && schema.type) || 'JSDoc'
```

Or make `type` required in rule schema.



---

Rule option example: 

```json
"rules": {
  "require-prop-comment": ["error", {}],
```

Reply:
ok